src/dbnode/persist/fs/files.go

```diff
@@ -1231,7 +1199,62 @@ func SortedCommitLogFiles(commitLogsDir string) ([]string, error) {
 	return sortedCommitLogFiles(commitLogsDir, commitLogFilePattern)
 }
 
+type filesetFile struct {
+	volumeIndex int
+	blockStart  xtime.UnixNano
+	fileName    string
+}
+
 type toSortableFn func(files []string) sort.Interface
+type toBlockStartAndVolumeIndexFn func(file string) (xtime.UnixNano, int, error)
+type sortedFilesetFiles []filesetFile
+
+func (s sortedFilesetFiles) Len() int {
+	return len(s)
+}
+
+func (s sortedFilesetFiles) Less(i, j int) bool {
+	ti := s[i].blockStart
+	tj := s[j].blockStart
+
+	if ti.Before(tj) {
+		return true
+	}
+
+	ij := s[j].volumeIndex
+	ii := s[i].volumeIndex
+	return ti.Equal(tj) && ii < ij
+}
+
+func (s sortedFilesetFiles) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+func findFilesetFiles(fileDir string, pattern string, fn toBlockStartAndVolumeIndexFn) (sortedFilesetFiles, error) {
```

Comment:
Nit: perhaps adding `Sorted` into the name of this function would make it more clear what it does.